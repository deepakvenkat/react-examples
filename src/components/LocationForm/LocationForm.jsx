import React from 'react';
import {
  TextField,
  Button,
} from '@material-ui/core';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  searchButton: {
    marginTop: '12px',
    marginLeft: '10px',
  }
};

const LocationForm = ({ classes }) =>  (
  <form>
    <TextField label="Zip Code" />
    <Button
      disableElevation
      variant="contained"
      className={classes.searchButton}
    >
      Search
    </Button>
  </form>
);

LocationForm.propTypes = {
  classes: PropTypes.shape({}).isRequired,
};

export default withStyles(styles)(LocationForm);
