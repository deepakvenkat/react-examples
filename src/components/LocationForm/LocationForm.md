## LocationForm

```jsx
<LocationForm
  searchZipcode={(zipcode) => console.log(zipcode)}
/>
```
