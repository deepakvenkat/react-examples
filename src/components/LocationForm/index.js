import LocationForm from './LocationForm.jsx';

export default LocationForm;
