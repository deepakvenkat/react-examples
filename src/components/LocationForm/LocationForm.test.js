import React from 'react';
import { render } from '@testing-library/react';
import LocationForm from './LocationForm';

test('renders the LocationForm', () => {
  const { asFragment } = render(
    <LocationForm />
  );
  expect(asFragment()).toMatchSnapshot();
});
