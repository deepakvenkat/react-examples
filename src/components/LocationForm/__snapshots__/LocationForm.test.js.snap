// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the LocationForm 1`] = `
<DocumentFragment>
  <form>
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
      >
        Zip Code
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained LocationForm-searchButton-1 MuiButton-disableElevation"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Search
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
